lib/connection.js

```diff
@@ -207,16 +207,16 @@ Connection.prototype._reconnect = function() {
         this.state.open();
     }
 
-    //reset transport
-    this.amqp_transport = new Transport(this.options.id, AMQP_PROTOCOL_ID, frames.TYPE_AMQP, this);
-    this.sasl_transport = undefined;
-    this.transport = this.amqp_transport;
-
     this.state.reconnect();
     this._reset_remote_state();
```

Comment:
I assume that resetting the transport is not a precondition to reconnecting the state. If that is indeed the case, then the current change would break reconnect. We can have a separate method named `_reset_transport()` then things here would look like:
```
this._reset_transport();
this.state.reconnect();
this._reset_remote_state();
```

and in the connect method

```
this._reset_transport();
this._reset_remote_state();
```